src/org/sosy_lab/java_smt/example/NQueens.java

```diff
@@ -0,0 +1,247 @@
+import org.sosy_lab.java_smt.SolverContextFactory.Solvers;
+import org.sosy_lab.java_smt.api.BooleanFormula;
+import org.sosy_lab.java_smt.api.BooleanFormulaManager;
+import org.sosy_lab.java_smt.api.Model;
+import org.sosy_lab.java_smt.api.ProverEnvironment;
+import org.sosy_lab.java_smt.api.SolverContext;
+import org.sosy_lab.java_smt.api.SolverContext.ProverOptions;
+import org.sosy_lab.java_smt.api.SolverException;
+public class NQueens {
+  public static void main(String... args)
+      throws InvalidConfigurationException, SolverException, InterruptedException {
+    Configuration config = Configuration.defaultConfiguration();
+    LogManager logger = BasicLogManager.create(config);
+    ShutdownNotifier notifier = ShutdownNotifier.createDummy();
+    Solvers solver = Solvers.SMTINTERPOL;
+    try (SolverContext context = SolverContextFactory.createSolverContext(config,
+        logger, notifier, solver)) {
+      NQueensSolver myQueen = new NQueen(context, 12);
+      Boolean[][] solutions = myQueen.solve(12);
+      if (solutions == null) {
+        System.out.println("No solutions found.");
+      } else {
+        System.out.println("Solution:");
+        for (Boolean[] pSolution : solutions) {
+          for (int col = 0; col < solutions[0].length; col++) {
+            if (pSolution[col]) {
+              System.out.print("Q ");
+            } else {
+              System.out.print("_ ");
+            }
+          }
+          System.out.println();
+        }
+        System.out.println();
+      }
+    } catch (InvalidConfigurationException | UnsatisfiedLinkError e) {
+      logger.logUserException(Level.INFO, e, "Solver " + solver + " is not available.");
+    } catch (UnsupportedOperationException e) {
+      logger.logUserException(Level.INFO, e, e.getMessage());
+    }
+  }
+
+  public abstract static class NQueensSolver {
+    private final SolverContext context;
+    final BooleanFormulaManager bmgr;
+
+    private NQueensSolver(SolverContext pContext) {
+      context = pContext;
+      bmgr = context.getFormulaManager().getBooleanFormulaManager();
+    }
+
+    abstract BooleanFormula[][] getSymbols();
+
+    abstract List<BooleanFormula> getRules(BooleanFormula[][] symbols);
+
+    abstract Boolean getValue(BooleanFormula[][] symbols, Model model, int row, int col);
+
+    /**
+     * Solves the N-Queens problem for the given board size and returns a possible solution.
+     * Returns <code>Null</code> if no solution exists.
+     */
+    public Boolean[][] solve(int n) throws InterruptedException, SolverException {
+      BooleanFormula[][] symbols = getSymbols();
+      List<BooleanFormula> rules = getRules(symbols);
+      // solve N-Queens
+      try (ProverEnvironment prover = context.newProverEnvironment(ProverOptions.GENERATE_MODELS)) {
+        prover.push(bmgr.and(rules));
+        boolean isUnsolvable = prover.isUnsat();
+        if (isUnsolvable) {
+          return new Boolean[0][0];
```

Comment:
there was "null" before. Please revert this line.
Example: This will not work for unsolvable sizes like N=2.

Reply:
I changed it because of these 2 errors
smt/src/org/sosy_lab/java_smt/example/NQueens.java:103:9: Method declares to return Boolean and returns null. [ReturnNullInsteadOfBoolean]
[WARN] /builds/sosy-lab/software/java-smt/src/org/sosy_lab/java_smt/example/NQueens.java:103:9: Method should return empty collection instead of null. [NoNullForCollectionReturn]

with the new code, I have declared a global variable of boolean array type that's always null and used it in the return statement, although I don't know if it would work or not @kfriedberger

Reply:
Seems like there is some error with the new commit, would try to correct it tomorrow morning if not for my classes as it's already too late now.

Reply:
you might need to annotate the function with `@Nullable`.